File: android development/Lab6_Chudakov/app/src/main/java/ru/chudakov/lab6_chudakov/MainActivity.java
package ru.chudakov.lab6_chudakov;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;

import android.os.Bundle;
import android.view.View;

import com.mikepenz.fontawesome_typeface_library.FontAwesome;
import com.mikepenz.materialdrawer.Drawer;
import com.mikepenz.materialdrawer.DrawerBuilder;
import com.mikepenz.materialdrawer.model.DividerDrawerItem;
import com.mikepenz.materialdrawer.model.PrimaryDrawerItem;
import com.mikepenz.materialdrawer.model.SecondaryDrawerItem;
import com.mikepenz.materialdrawer.model.interfaces.IDrawerItem;

import ru.chudakov.lab6_chudakov.home.HomeFragment;
import ru.chudakov.lab6_chudakov.info.InfoFragment;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Toolbar toolbar = findViewById(R.id.tool_bar);

        final PrimaryDrawerItem home = new PrimaryDrawerItem().withName(R.string.menu_home);
        final PrimaryDrawerItem info = new PrimaryDrawerItem().withName(R.string.menu_info);

        final Fragment homeFragment = new HomeFragment();
        final Fragment infoFragment = new InfoFragment();

        changeFragment(homeFragment);

        new DrawerBuilder()
                .withActivity(this)
                .withToolbar(toolbar)
                .addDrawerItems(
                        home
                )
                .withOnDrawerItemClickListener(new Drawer.OnDrawerItemClickListener() {
                    @Override
                    public boolean onItemClick(View view, int position, IDrawerItem drawerItem) {
                        Fragment fragment;

                        switch (position) {
                            case 0: {
                                fragment = homeFragment;
                                break;
                            }
                            case 1: {
                                fragment = homeFragment;
                                break;
                            }
                            case 3: {
                                fragment = homeFragment;
                                break;
                            }
                            case 4: {
                                fragment = infoFragment;
                                break;
                            }
                            default: {
                                return true;
                            }
                        }
                        changeFragment(fragment);

                        return false;
                    }
                })
                .build();
    }

    private void changeFragment(Fragment fragment) {
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.container, fragment)
                .commit();
    }
}

File: android-development/Portfolio_Chudakov/app/src/main/java/ru/chudakov/portfolio_chudakov/MainActivity.kt
package ru.chudakov.portfolio_chudakov

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.mikepenz.materialdrawer.AccountHeaderBuilder
import com.mikepenz.materialdrawer.DrawerBuilder
import com.mikepenz.materialdrawer.model.DividerDrawerItem
import com.mikepenz.materialdrawer.model.PrimaryDrawerItem
import com.mikepenz.materialdrawer.model.SecondaryDrawerItem
import kotlinx.android.synthetic.main.app_bar_main.*
import ru.chudakov.portfolio_chudakov.aboutMe.AboutMeFragment
import ru.chudakov.portfolio_chudakov.contacts.ContactsFragment
import ru.chudakov.portfolio_chudakov.job.JobFragment
import ru.chudakov.portfolio_chudakov.plans.PlansFragment
import ru.chudakov.portfolio_chudakov.studing.StudyFragment

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val header = AccountHeaderBuilder()
            .withActivity(this)
            .withHeaderBackground(R.drawable.cat)
            .build()

        val aboutMeItem = PrimaryDrawerItem().withName(R.string.about_me)
        val studyItem = PrimaryDrawerItem().withName(R.string.study)
        val jobItem = PrimaryDrawerItem().withName(R.string.job)
        val plansItem = PrimaryDrawerItem().withName(R.string.plans)
        val contactsItem = SecondaryDrawerItem().withName(R.string.contacts)

        val aboutMeFragment = AboutMeFragment()
        val studyFragment = StudyFragment()
        val jobFragment = JobFragment()
        val plansFragment = PlansFragment()
        val contactsFragment = ContactsFragment()

        changeFragment(aboutMeFragment)

        DrawerBuilder()
            .withActivity(this)
            .withToolbar(toolBar)
            .withAccountHeader(header)
            .addDrawerItems(
                aboutMeItem,
                studyItem,
                jobItem,
                plansItem,
                DividerDrawerItem(),
                contactsItem
            )
            .withOnDrawerItemClickListener { _, position, _ ->
                var result = false;

                when (position) {
                    1 -> changeFragment(aboutMeFragment)
                    2 -> changeFragment(studyFragment)
                    3 -> changeFragment(jobFragment)
                    4 -> changeFragment(plansFragment)
                    6 -> changeFragment(contactsFragment)
                    else -> result = true;
                }

                result
            }
            .build()
    }

    private fun changeFragment(fragment: Fragment) {
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.container, fragment)
            .commit()
    }
}
